### PR TITLE
Connect to real API - BW-582

### PR DESCRIPTION
**Pull request by dgmdan**
Resolves BW-582. Some miscellaneous improvements in here too.

### CODE REVIEW

**Comment by eugenebiz** [comment]
Dan, if you're connecting to the actual API here then you'll need the "format=json" URL parameter tacked onto the various locations you're using the API url.

**Comment by dgmdan** [comment]
Updated.. although it was returning json just fine without the parameter

**Comment by eugenebiz** [comment]
Interesting. When going to the url through the browser it returns their decorated page. I wonder if the call from php is setting a header or something?

File: Bizyhood/Views/listings/single/default.php [hunk 1]
```diff
@@ -1,144 +1,90 @@
...
-        <?php if($meta['bs_address_1']): ?>
-        <?php echo Broadstreet_Utility::buildAddressFromMeta($meta); ?>
+        <?php if($business->address1): ?>
+            <?php echo Broadstreet_Utility::buildAddressFromMeta($business); ?>
         <?php endif; ?>
         <br />
-        <?php if($meta['bs_address_1']): ?>
-        <a target="_blank" href="https://maps.google.com/?q=<?php echo urlencode(Broadstreet_Utility::buildAddressFromMeta($meta, true)) ?>">View map</a><br />
+        <?php if($business->address1): ?>
+            <a target="_blank" href="https://maps.google.com/?q=<?php echo urlencode(Broadstreet_Utility::buildAddressFromMeta($business, true)) ?>">View map</a><br />
```
Comment (eugenebiz) [inline Bizyhood/Views/listings/single/default.php]: the protocol should be dropped and left as a relative url e.g. //maps.google.com....

File: Bizyhood/Utility.php [hunk 2]
```diff
@@ -95,6 +89,42 @@ public static function getApiKey()
...
+    /**
+     * Get the Bizyhood API URL
+     * @return string 
+     */
+    public static function getApiUrl()
+    {
+        $api_url = Broadstreet_Utility::getOption(Bizyhood_Core::KEY_API_URL);
+        
+        if(!$api_url || strlen($api_url) == 0)
+            return "https://bizyhood.com/api";
```
Comment (eugenebiz) [inline Bizyhood/Utility.php]: A default doesn't make sense. This should always be configured in settings. How do we ensure that the settings are set?

File: Bizyhood/Views/listings/single/default.php (hunk 1 above)
Comment (dgmdan) [inline Bizyhood/Views/listings/single/default.php]: Updated

File: Bizyhood/Utility.php (hunk 2 above)
Comment (dgmdan) [inline Bizyhood/Utility.php]: My thought was that the default would be the real API URL. Won't you
eventually want to hide this setting entirely so people don't mess with it?

Comment (eugenebiz) [inline Bizyhood/Utility.php]: Actually, no. I anticipate having to have experimental endpoints for early adopters, etc. I'd like to keep this explicit for now. Additionally, the default you chose is invalid.

Comment (dgmdan) [inline Bizyhood/Utility.php]: Okay, updated so there's no default.

File: Bizyhood/Views/listings/single/default.php (hunk 1 above)
Comment (eugenebiz) [inline Bizyhood/Views/listings/single/default.php]: Protocol is still there.

**Comment by eugenebiz** [comment]
Ship it.

File: Bizyhood/Views/listings/single/default.php (hunk 1 above)
Comment (dgmdan) [inline Bizyhood/Views/listings/single/default.php]: It's fixed on master now. Not sure what happened, perhaps I committed to the other PR.

**Pull request merged**